### PR TITLE
ensuarance  api modified which contains PodQOSEnsurancePolicy ,NodeEnsurancePolicy and AvoidanceAction

### PR DESCRIPTION
**Pull request by yan234280533**

### CODE REVIEW

File: ensurance/v1alpha1/types.go [hunk 1]
```diff
@@ -153,3 +210,96 @@ type NodeLocalGet struct {
...
+	CPUThrottle CPUThrottle `json:"cpuThrottle,omitempty"`
+
+	// +optional
+	MemoryThrottle MemoryThrottle `json:"memoryThrottle,omitempty"`
+}
+
+type CPUThrottle struct {
+	// how long it waits for each compress step
+	// +optional
+	Interval time.Duration `json:"interval,omitempty"`
```
Comment (qmhu) [inline ensurance/v1alpha1/types.go:225]: Interval -> IntervalSeconds

File: ensurance/v1alpha1/types.go [hunk 2]
```diff
@@ -153,3 +210,96 @@ type NodeLocalGet struct {
...
+
+	//the step of cpu share and limit for once down-size (1-100)
+	// +optional
+	StepCPURatio uint64 `json:"stepCPURatio,omitempty"`
+}
+
+type MemoryThrottle struct {
+	// how long it waits for each compress step
+	// +optional
+	Interval time.Duration `json:"interval,omitempty"`
```
Comment (qmhu) [inline ensurance/v1alpha1/types.go:240]: The same above

File: ensurance/v1alpha1/types.go (hunk 1 above)
Comment (yan234280533) [reply ensurance/v1alpha1/types.go]: I modified it in the new version

File: ensurance/v1alpha1/types.go (hunk 2 above)
Comment (yan234280533) [reply ensurance/v1alpha1/types.go]: I modified it in the new version

File: ensurance/v1alpha1/types.go [hunk 3]
```diff
@@ -148,8 +213,112 @@ type NodeHandler struct {
...
+	IntervalSeconds *int32 `json:"intervalSeconds,omitempty"`
+
+	//the min of cpu ratio for low level pods
+	//example: the pod limit is 4096, ratio is 10, the min is 409
+	// +optional
+	MinCPURatio uint64 `json:"minCPURatio,omitempty"`
+
+	//the step of cpu share and limit for once down-size (1-100)
+	// +optional
+	StepCPURatio uint64 `json:"stepCPURatio,omitempty"`
```
[the remark this answers is not in the document]
Comment (yan234280533) [reply ensurance/v1alpha1/types.go:245]: done~

File: ensurance/v1alpha1/types.go [hunk 4]
```diff
@@ -148,8 +213,112 @@ type NodeHandler struct {
...
 
 type NodeLocalGet struct {
+	// default is 60s
 	// +optional
-	LocalCacheTTL time.Duration `json:"localCacheTTL,omitempty"`
+	LocalCacheTTLSeconds *int64 `json:"localCacheTTLSeconds,omitempty"`
+	// default is 60s
 	// +optional
-	MaxHousekeepingInterval time.Duration `json:"maxHousekeepingInterval,omitempty"`
+	MaxHousekeepingIntervalSeconds *int64 `json:"maxHousekeepingIntervalSeconds,omitempty"`
```
[the remark this answers is not in the document]
Comment (yan234280533) [reply ensurance/v1alpha1/types.go:221]: done~

File: ensurance/v1alpha1/types.go [hunk 5]
```diff
@@ -36,6 +35,9 @@ type PodQOSEnsurancePolicySpec struct {
 
 	//pod quality probe
 	QualityProbe QualityProbe `json:"qualityProbe,omitempty"`
+
+	//pod objective ensurance check and action
```
[the remark this answers is not in the document]
Comment (yan234280533) [reply ensurance/v1alpha1/types.go:39]: done~

File: ensurance/v1alpha1/types.go [hunk 6]
```diff
@@ -85,6 +92,56 @@ type NodeQOSEnsurancePolicyList struct {
...
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AvoidanceActionSpec   `json:"spec"`
+	Status AvoidanceActionStatus `json:"status,omitempty"`
+}
+
+type AvoidanceActionSpec struct {
+	// how long it should wait between blocking/unblocking scheduling
+	SchedulingCoolDown time.Duration `json:"schedulingCoolDown,omitempty"`
```
[the remark this answers is not in the document]
Comment (yan234280533) [reply ensurance/v1alpha1/types.go]: done~

File: ensurance/v1alpha1/types.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (yan234280533) [reply ensurance/v1alpha1/types.go:39]: I will modify these in the other pr.

**Pull request merged**